packages/mds-ingest-service/repository/entities/event-entity.ts

```diff
@@ -42,7 +42,7 @@ export class EventEntity extends IdentityColumn(RecordedColumn(class {})) implem
   provider_id: EventEntityModel['provider_id']
 
   @Column('bigint', { transformer: BigintTransformer, primary: true })
-  timestamp: EventEntityModel['timestamp']
+  timestamp: Timestamp
```

Comment:
Strange type behavior here with the `typeorm-cursor-pagination` lib.
When this is left as `EventEntityModel['timestamp']`, and pass `limit` to the `buildPaginator(...)` function, it results in the following error:

```
unknown type in cursor: [object]1625149089973
```

Which corresponds to the `time_range: {begin}` parameter.

Interestingly, this is only an issue when limit is passed. If the type declaration here is left as `EventEntityModel['timestamp']` and _limit_ **isn't** provided, it works fine. As soon as _limit_ is provided, it breaks.

Changing this to `Timestamp` directly here fixes the issue in both cases.

This doesn't have any issues afaict, all previous tests are still passing with this change. 🤷

Reply:
We've recently stopped defining entity properties in terms of Entity or (especially) Domain model properties and use the base types instead. Some libraries that use metadata reflection have issues tracing the aliases to the base type. In recent repositories, you will see a pattern something like:

```ts
export class SomeEntity {
  @Column(...)
  entity_id: UUID

  @Column(...)
  timestamp: Timestamp
}

/* Create an alias to keep consistent with other services */
export SomeEntityModel = SomeEntity
```